PreFreeSurfer/PreFreeSurferPipeline.sh

```diff
@@ -452,253 +517,312 @@ log_Msg "POSIXLY_CORRECT="${POSIXLY_CORRECT}
+        --t2sampspacing=${T2wSampleSpacing} \
+        --unwarpdir=${UnwarpDir} \
+        --ot1=${T1wFolder}/${T1wImage}_acpc_dc \
+        --ot1brain=${T1wFolder}/${T1wImage}_acpc_dc_brain \
+        --ot1warp=${T1wFolder}/xfms/${T1wImage}_dc \
+        --ot2=${T1wFolder}/${T2wImage}_acpc_dc \
+        --ot2warp=${T1wFolder}/xfms/${T2wImage}_reg_dc \
+        --method=${AvgrdcSTRING} \
+        --topupconfig=${TopupConfig} \
+        --gdcoeffs=${GradientDistortionCoeffs} \
+        --usejacobian=${UseJacobian} 
+
+      ;;
+
+    *)
+
+      log_Msg "NOT PERFORMING READOUT DISTORTION CORRECTION"
+      wdir=${T2wFolder}/T2wToT1wReg
+      if [ -e ${wdir} ] ; then
+        # DO NOT change the following line to "rm -r ${wdir}" because the
+        # chances of something going wrong with that are much higher, and
+        # rm -r always needs to be treated with the utmost caution
+        rm -r ${T2wFolder}/T2wToT1wReg
+      fi
+
+      log_Msg "mkdir -p ${wdir}"
+      mkdir -p ${wdir}
+
+      ${RUN} ${HCPPIPEDIR_PreFS}/T2wToT1wReg.sh \
+        ${wdir} \
+        ${T1wFolder}/${T1wImage}_acpc \
+        ${T1wFolder}/${T1wImage}_acpc_brain \
+        ${T2wFolder_T2wImageWithPath_acpc} \
+        ${T2wFolder_T2wImageWithPath_acpc_brain} \
+        ${T1wFolder}/${T1wImage}_acpc_dc \
+        ${T1wFolder}/${T1wImage}_acpc_dc_brain \
+        ${T1wFolder}/xfms/${T1wImage}_dc \
+        ${T1wFolder}/${T2wImage}_acpc_dc \
+        ${T1wFolder}/xfms/${T2wImage}_reg_dc 
+
+  esac
+
+  # ------------------------------------------------------------------------------
+  #  Bias Field Correction: Calculate bias field using square root of the product
+  #  of T1w and T2w iamges.
+  # ------------------------------------------------------------------------------
+
+  if [ ! "${T2wInputImages}" = "NONE" ] ; then
+
+    log_Msg "Performing Bias Field Correction"
+    if [ ! -z ${BiasFieldSmoothingSigma} ] ; then
+      BiasFieldSmoothingSigma="--bfsigma=${BiasFieldSmoothingSigma}"
+    fi
+
+    log_Msg "mkdir -p ${T1wFolder}/BiasFieldCorrection_sqrtT1wXT1w"
+    mkdir -p ${T1wFolder}/BiasFieldCorrection_sqrtT1wXT1w
+
+    ${RUN} ${HCPPIPEDIR_PreFS}/BiasFieldCorrection_sqrtT1wXT1w.sh \
+      --workingdir=${T1wFolder}/BiasFieldCorrection_sqrtT1wXT1w \
+      --T1im=${T1wFolder}/${T1wImage}_acpc_dc \
+      --T1brain=${T1wFolder}/${T1wImage}_acpc_dc_brain \
+      --T2im=${T1wFolder_T2wImageWithPath_acpc_dc} \
+      --obias=${T1wFolder}/BiasField_acpc_dc \
+      --oT1im=${T1wFolder}/${T1wImage}_acpc_dc_restore \
+      --oT1brain=${T1wFolder}/${T1wImage}_acpc_dc_restore_brain \
+      --oT2im=${T1wFolder}/${T2wImage}_acpc_dc_restore \
+      --oT2brain=${T1wFolder}/${T2wImage}_acpc_dc_restore_brain \
+      ${BiasFieldSmoothingSigma}
+
+  else  # -- No T2w image
+
+    log_Msg "Performing Bias Field Correction on T1w image only"
+
+    if [ ! -z ${BiasFieldSmoothingSigma} ] ; then
+      BiasFieldSmoothingSigma="--bfsigma=${BiasFieldSmoothingSigma}"
+    fi
```

Comment:
If we wanted, we could issue this conditional just once, immediately preceding the `if [ ! "${T2wInputImages}" = "NONE" ] ; then` conditional.

Reply:
indeed. done